### PR TITLE
Rename HostDir to HostPath in v1beta3

### PR DESCRIPTION
**Pull request by thockin**
I started with HostMount (as can still be seen in the volumes PR), but I like this better.  It's a minor point, but it irks me.  This could also be done in a compatible way, and accept both HostDir and HostPath, if anyone thinks that is worthwhile.  I think it might be.

### CODE REVIEW

**Comment by bgrant0607** [comment]
LGTM

**Comment by thockin** [comment]
Is it worth accepting both HostDir and HostPath?  To do so I need to
bootstrap the v1beta3 conversions.go...

**Comment by thockin** [comment]
Actually, I think this can be simpler.

**Comment by bgrant0607** [comment]
You mean you're planning to simplify the conversion code?

**Comment by thockin** [comment]
Yes, but I am OOO, so please do not merge this yet.

**Comment by thockin** [comment]
updated.  Conversions are a smidge simpler and v1beta3 accepts either hostDir or hostPath.

**Comment by thockin** [comment]
Rebased, Travis is running.

File: pkg/api/v1beta3/types.go [hunk 1]
```diff
@@ -170,12 +170,14 @@ type Volume struct {
...
-	// things that are allowed to see the host machine. Most containers will NOT need this.
+	// HostPath represents a pre-existing file or directory on the host
+	// machine that is directly exposed to the container. This is generally
+	// used for system agents or other privileged things that are allowed
+	// to see the host machine. Most containers will NOT need this.
 	// TODO(jonesdl) We need to restrict who can use host directory mounts and who can/can not
 	// mount host directories as read/write.
-	HostDir *HostDir `json:"hostDir"`
+	HostPath *HostPath `json:"hostPath"`
+	HostDir  *HostPath `json:"hostDir"` // DEPRECATED: use HostPath instead
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: Could we not start v1beta3 with deprecated fields? Why is this needed?

Comment (thockin) [inline pkg/api/v1beta3/types.go]: The main Volumes[] structure has not changed at all between v1beta1 and
v1beta3.  I thought it was fair to allow the old name.  we don't have to,
but I expect a bit of pain

Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: If we believe the change is compelling, let's just do it. If not, let's leave it as is. Most users shouldn't be using this field, anyway.

Comment (thockin) [inline pkg/api/v1beta3/types.go]: It's a minor point, but I have been asked a few times whether HostDir
supported files, too.  It does -> rename.  I'll drop the compat stuff if
you think compat is not so compelling (though I find the cost to be minor).

Say yes one more time, and I will do it :)

Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: Yes, drop compat. Thanks.

Also, please update the description text in v1beta1/2 to reflect that it also supports files. I didn't know that, either.

**Comment by thockin** [comment]
Now with less compatibility!

**Comment by bgrant0607** [comment]
LGTM
